src/twisted/test/test_tcp.py

```diff
@@ -187,6 +187,21 @@ def test_listen(self):
         self.addCleanup(p1.stopListening)
         self.assertTrue(interfaces.IListeningPort.providedBy(p1))
 
+    def test_multiListen(self):
+        """
+        Test that multiple sockets can listen on the same port
+        """
+        if not hasattr(socket, "SO_REUSEPORT"):
+            raise unittest.SkipTest('SO_REUSEPORT not supported')
```

Comment:
My understading is that on Windows, SO_REUSEADDR also implements SO_REUSEPORT... so listenMultiple is also supported on Windows.

But on Windows, we might want to use SO_EXCLUSIVEADDRUSE

https://docs.microsoft.com/en-us/windows/win32/winsock/using-so-reuseaddr-and-so-exclusiveaddruse?redirectedfrom=MSDN

Maybe this test should be extend by making 100 ping connections and check that both factories will receive at least one connection.

We don't know if the OS will do a round-robin between the ports, but I guess that our of 100 connection we should get at least another connection.

At the same time, this is not testing listening to the same port from different twisted processes ... but I guess that this is something for OS and the current level of testing is OK for Twisted. 

Also, I think that we need a test in which when listenMultiple=False we see that  the second listenTCP fails... just to make sure we don't accidentally set listenMultiple all the time